ckanext/geocat/tests/test_distribution_metadata.py

```diff
@@ -35,81 +36,78 @@ def _load_xml(self, filename):
+        self.assertSetEqual({'de', 'fr', 'en', 'it'}, set(distribution.get('language')))
+        self.assertEquals('http://www.bafu.admin.ch/umwelt/12877/15716/15721/index.html?lang=de', distribution.get('url'))
+        self.assertEquals('http://www.bafu.admin.ch/umwelt/12877/15716/15721/index.html?lang=de', distribution.get('download_url'))
+        self.assertEquals(distribution.get('url'), distribution.get('download_url'))
+        self.assertEquals('NonCommercialAllowed-CommercialAllowed-ReferenceNotRequired', distribution.get('rights'))
+        self.assertEquals('', distribution.get('media_type'))
+
+    def test_normed_protocol_WWW_DOWNLOAD_INTERLIS(self):
+        xml = self._load_xml('geocat-testdata.xml')
+        dataset = self.csw_map.get_metadata(xml, self.geocat_identifier_normed_protocols)
+        distributions = dataset.get('resources')
+        distribution = _get_distribution_by_protocol(distributions, "WWW:DOWNLOAD:INTERLIS")
+        self.assertEquals('https://data.geo.admin.ch', distribution.get('url'))
+        self.assertEquals('https://data.geo.admin.ch', distribution.get('download_url'))
+        self.assertEquals(distribution.get('url'), distribution.get('download_url'))
+        self.assertEquals('NonCommercialAllowed-CommercialAllowed-ReferenceRequired', distribution.get('rights'))
+        self.assertEquals('INTERLIS', distribution.get('media_type'))
+
+    def test_normed_protocol_WWW_DOWNLOAD_APP(self):
+        xml = self._load_xml('geocat-testdata.xml')
+        dataset = self.csw_map.get_metadata(xml, self.geocat_identifier_normed_protocols)
+        distributions = dataset.get('resources')
+        distribution = _get_distribution_by_protocol(distributions, "WWW:DOWNLOAD-APP")
+        self.assertEquals('https://geocat.geoshop.ch', distribution.get('url'))
+        self.assertIsNone(distribution.get('download_url'))
 
-        # access url + download url
-        self.assertEquals('http://data.geo.admin.ch/ch.bafu.laerm-bahnlaerm_nacht/data.zip', download.get('url'))
-        self.assertEquals('http://data.geo.admin.ch/ch.bafu.laerm-bahnlaerm_nacht/data.zip', download.get('download_url'))
-        self.assertEquals(download.get('url'), download.get('download_url'))
+    def test_normed_protocol_Map_Preview(self):
+        xml = self._load_xml('geocat-testdata.xml')
+        dataset = self.csw_map.get_metadata(xml, self.geocat_identifier_normed_protocols)
+        distributions = dataset.get('resources')
+        distribution = _get_distribution_by_protocol(distributions, "MAP:Preview")
+        self.assertEquals('https://map.geo.admin.ch/?layers=ch.bfe.energiestaedte', distribution.get('url'))
+        self.assertIsNone(distribution.get('download_url'))
 
-        # rights
-        self.assertEquals('NonCommercialAllowed-CommercialAllowed-ReferenceNotRequired', download.get('rights')) 
+    def test_normed_protocol_OGC_WMS(self):
```

Comment:
in general, all tests are have mostly the same structure. I am not sure, but would it make sense to use a common body (class template) and just change attributes that are specific to the test..
If it is to much work, I will leave it like it is